### ISSUE TITLE
Builds for Z in OperatorHub

### ISSUE THREAD

**Issue by cfilleke** [OpenLiberty/open-liberty-operator#121]
## Feature Request

To have regular builds for Z provided in the OperatorHub for test and use in OpenShift 4.3+ on Z 

### Is your feature request related to a problem?

Open Liberty Operator does not currently install or operate on s390x architecture through OpenShift/OperatorHub 

### Describe the solution you'd like

A way to provide Metadata for Operators to indicate the architecture of the build (e.g. amd64, s390x, ppc64le...) is being worked out between the Red Hat Multi-Arch team and OperatorHub.io -- we'd hoped it would show up in OpenShift 4.3 for z/VM, but this does not seem to be the case. 

In any case, I think the place to start would be to add s390x to the list of architectures Travis is building for each pull request.

**Comment by leochr** [comment]
@cherylaf Please checkout the multi-arch images (including Z) for Open Liberty Operator available at https://hub.docker.com/r/openliberty/operator/tags

**Comment by cherylaf** [comment]
Thanks, @leochr !  

OpenShift does not seem to be accessing these images when subscribed to the OperatorHub -- It looks like it's accessing the wrong registry, and trying to pull in the wrong arch image:

```
Events:
  Type     Reason     Age                     From                                      Message
  ----     ------     ----                    ----                                      -------
  Normal   Scheduled  <unknown>               default-scheduler                         Successfully assigned cpu-stress/open-liberty-operator-65fdc75df4-25qns to worker-1.pok-feb26.zvmocp.notld
  Normal   Pulling    5m55s (x4 over 7m22s)   kubelet, worker-1.pok-feb26.zvmocp.notld  Pulling image "registry.connect.redhat.com/ibm/open-liberty-operator:0.3.0"
  Warning  Failed     5m52s (x4 over 7m20s)   kubelet, worker-1.pok-feb26.zvmocp.notld  Failed to pull image "registry.connect.redhat.com/ibm/open-liberty-operator:0.3.0": rpc error: code = Unknown desc = Image architecture mismatch: image uses "amd64", expecting "s390x"
  Warning  Failed     5m52s (x4 over 7m20s)   kubelet, worker-1.pok-feb26.zvmocp.notld  Error: ErrImagePull
  Normal   BackOff    5m27s (x6 over 7m19s)   kubelet, worker-1.pok-feb26.zvmocp.notld  Back-off pulling image "registry.connect.redhat.com/ibm/open-liberty-operator:0.3.0"
  Warning  Failed     2m12s (x20 over 7m19s)  kubelet, worker-1.pok-feb26.zvmocp.notld  Error: ImagePullBackOff
```

**Comment by arthurdm** [comment]
hi @cherylaf - the Open Liberty Operator version inside OCP is our certified version, so it uses a different registry than our OperatorHub.io version (in order to pass the certification scan).

The blocking issue is that the Red Hat certification portal does not support "manifest list" which could proxy over multi-architectures builds.  Docker Hub does this, but Red Hat registry does not.

**Comment by cherylaf** [comment]
hi @arthurdm -- thanks for the information! 

Some operators are being published with a separate subscription channel for s390x as a work-around until the multi-arch metadata is fully supported by operatorhub.io.   https://bugzilla.redhat.com/show_bug.cgi?id=1766278 indicates that OCP 4.4 would be the earliest version we could expect that in. 

![image](https://user-images.githubusercontent.com/3160773/77099558-86059d00-69ea-11ea-8b6d-e640f28827b7.png)  

Are we sure the certification portal is still munted with respect to multi-arch?

**Comment by arthurdm** [comment]
Thanks @cherylaf - we could look into the subscription approach, thanks for sharing that.  I am curious though: once a particular channel is chosen (let's say `s390x`) how does OLM ensure only Nodes with that architecture are used, in the case of a hybrid node architecture?   So for example, not deploying a `s390x` controller into a x86 Node.

From what I understood from the folks at the Red Hat portal, the issue is that OpenShift's Registry does not support multi-arch manifest lists - and the portal uses an OCP Registry in the background. (instead of quay).

**Comment by cherylaf** [comment]
The question came up in Westford a couple weeks ago at the Multi-Arch RHPE training when hybrid OCP cluster support would be available, and the answer was 'not for a long time'.  The tag list in the 2nd screen shot above is from the registry, which does have multi-arch support now.  

Here's a short animation of first accessing a multi-arch operator image in the registry, and then accessing the open liberty multi-arch operator in the registry:
  
![ani-reg](https://user-images.githubusercontent.com/3160773/77119103-2ae4a200-6a0c-11ea-8bde-7912957de911.gif)

**Comment by arthurdm** [comment]
hey @cherylaf  - I can successfully upload (and I presume publish too) our multi-arch images via the Red Hat Portal:

<img width="624" alt="Screen Shot 2020-03-20 at 11 50 12 AM" src="https://user-images.githubusercontent.com/5042215/77180972-130b2d80-6aa1-11ea-8f47-6c63f223cc96.png">

But when I try to push a `manifest-list` that would allow users to proxy over those images with the same `docker pull` or `FROM` value, I get an error:

```
Registry push unsuccessful: response 400: 400 Bad Request

**Comment by arthurdm** [comment]
`

**Comment by cherylaf** [comment]
Sending y'all IBM email to follow up, since it contains RH/IBM confidential info.

**Comment by leochr** [comment]
Support for Z (s390x) is available from Open Liberty Operator 1.2.0+

Closing as completed
